### PR TITLE
chore: [SFEQS-1419] Update signed message to support third party content

### PR DESCRIPTION
**Pull request by grausof**
This PR adds support for `third party content` and modifies the `message` of successful signature.

#### List of Changes
- Added `third_party_data` as an optional parameter in `@io-sign/io-sign/notification`
- Updated message of successful signature

#### How Has This Been Tested?
Tested locally with cloud resources

#### Checklist:
- [ ] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/io-sign/src/infra/io-services/message.ts [hunk 1]
```diff
@@ -36,6 +37,11 @@ export const makeSubmitMessageForUser =
               message: {
                 ...message,
                 fiscal_code: fiscalCode,
+                /* feature_level_type field is used to identify the institutions that have subscribed to premium messages.
+                 * In our case we have not adhered to any agreement therefore the field remains STANDARD but
+                 * in any case we are enabled to use the attachments feature.
```
Comment (gunzip) [inline packages/io-sign/src/infra/io-services/message.ts:42]: how?

**Review by lucacavallaro — CHANGES_REQUESTED** [review]

File: packages/io-sign/src/infra/io-services/message.ts (hunk 1 above)
Comment (gunzip) [reply packages/io-sign/src/infra/io-services/message.ts:42]: I'd add this to the comment

File: packages/io-sign/src/notification.ts [hunk 2]
```diff
@@ -11,12 +12,23 @@ export const Notification = t.type({
...
     subject: t.string,
     markdown: t.string,
   }),
+  t.partial({
+    third_party_data: t.type({
+      id: NonEmptyString,
+      has_attachments: t.boolean,
+    }),
+  }),
+]);
```
[the remark this answers is not in the document]
Comment (grausof) [reply packages/io-sign/src/notification.ts:26]: Yes, I was thinking about doing a refactor like this. And then maybe with an encoder on the infra side. I proceed and ask you for a review!

File: packages/io-sign/src/infra/io-services/message.ts [hunk 3]
```diff
@@ -15,6 +19,25 @@ import { ActionNotAllowedError, TooManyRequestsError } from "../../error";
...
+export const NotificationMessageToApiModel: Enc.Encoder<
+  NewMessage,
+  NotificationMessage
+> = {
+  encode: (message) => ({
+    content: {
+      subject: message.subject,
+      markdown: message.markdown,
+      third_party_data:
+        "signatureRequestId" in message
```
Comment (gunzip) [inline packages/io-sign/src/infra/io-services/message.ts:31]: What about having two different encoders instead? 
(`NotificationContent` vs `NotificationContentWithAttachments`)

Comment (grausof) [reply packages/io-sign/src/infra/io-services/message.ts]: Done! it's a good idea!

File: packages/io-sign/src/notification.ts [hunk 4]
```diff
@@ -11,13 +12,27 @@ export const Notification = t.type({
...
+  t.type({
+    signatureRequestId: Id,
+  }),
+]);
+
+export type NotificationContentWithAttachments = t.TypeOf<
+  typeof NotificationContentWithAttachments
+>;
+
+export type NotificationMessage =
```
Comment (gunzip) [inline packages/io-sign/src/notification.ts:33]: if you use a io-ts union instead you can call `NotificationMessage.encode` without first checking for the `signatureId`.

Comment (grausof) [reply packages/io-sign/src/notification.ts:33]: It's not clear to me. How do I define more than one encoder for the same type? NotificationMessage is part of the io-sign domain

Comment (gunzip) [reply packages/io-sign/src/notification.ts:33]: you're right sorry :-)

**Review by lucacavallaro — APPROVED** [review]

**Pull request merged**